src/main/drivers/barometer/barometer_qmp6988.c

```diff
@@ -0,0 +1,304 @@
+#include "drivers/bus.h"
+#include "drivers/bus_i2c.h"
+#include "drivers/bus_i2c_busdev.h"
+#include "drivers/bus_spi.h"
+#include "drivers/io.h"
+#include "drivers/time.h"
+
+#include "drivers/serial.h"
+#include "io/serial.h"
+#include "common/printf.h"
+#include "barometer_qmp6988.h"
+
+#if defined(USE_BARO) && (defined(USE_BARO_QMP6988) || defined(USE_BARO_SPI_QMP6988))
+
+typedef struct qmp6988_calib_param_s {
+	float Coe_a0;
+	float Coe_a1;
+	float Coe_a2;
+	float Coe_b00;
+	float Coe_bt1;
+	float Coe_bt2;
+	float Coe_bp1;
+	float Coe_b11;
+	float Coe_bp2;
+	float Coe_b12;
+	float Coe_b21;
+	float Coe_bp3;
+} qmp6988_calib_param_t;
+
+static uint8_t qmp6988_chip_id = 0;
+STATIC_UNIT_TESTED qmp6988_calib_param_t qmp6988_cal;
+// uncompensated pressure and temperature
+int32_t qmp6988_up = 0;
+int32_t qmp6988_ut = 0;
+
+static void qmp6988_start_ut(baroDev_t *baro);
+static void qmp6988_get_ut(baroDev_t *baro);
+static void qmp6988_start_up(baroDev_t *baro);
+static void qmp6988_get_up(baroDev_t *baro);
+
+STATIC_UNIT_TESTED void qmp6988_calculate(int32_t *pressure, int32_t *temperature);
+
+
+void qmp6988BusInit(busDevice_t *busdev)
+{
+#ifdef USE_BARO_SPI_QMP6988
+    if (busdev->bustype == BUSTYPE_SPI) {
+    	IOHi(busdev->busdev_u.spi.csnPin); // Disable
+        IOInit(busdev->busdev_u.spi.csnPin, OWNER_BARO_CS, 0);
+        IOConfigGPIO(busdev->busdev_u.spi.csnPin, IOCFG_OUT_PP);
+        spiSetDivisor(busdev->busdev_u.spi.instance, SPI_CLOCK_STANDARD); // XXX
+    }
+#else
+    UNUSED(busdev);
+#endif
+}
+
+void qmp6988BusDeinit(busDevice_t *busdev)
+{
+#ifdef USE_BARO_SPI_QMP6988
+    if (busdev->bustype == BUSTYPE_SPI) {
+        IOConfigGPIO(busdev->busdev_u.spi.csnPin, IOCFG_IPU);
+        IORelease(busdev->busdev_u.spi.csnPin);
+        IOInit(busdev->busdev_u.spi.csnPin, OWNER_SPI_PREINIT, 0);
+    }
+#else
+    UNUSED(busdev);
+#endif
+}
+
+bool qmp6988Detect(baroDev_t *baro)
+{
+    uint8_t databuf[25] = {0};
+    int Coe_a0_;
+    int Coe_a1_;
+    int Coe_a2_;
+    int Coe_b00_;
+    int Coe_bt1_;
+    int Coe_bt2_;
+    int Coe_bp1_;
+    int Coe_b11_;
+    int Coe_bp2_;
+    int Coe_b12_;
+    int Coe_b21_;
+    int Coe_bp3_;
+    u16 lb=0,hb=0;
+    u32 lw=0,hw=0,temp1,temp2;
+
+    delay(20);
+
+    busDevice_t *busdev = &baro->busdev;
+    bool defaultAddressApplied = false;
+
+    qmp6988BusInit(busdev);
+
+    if ((busdev->bustype == BUSTYPE_I2C) && (busdev->busdev_u.i2c.address == 0)) {
+        busdev->busdev_u.i2c.address = QMP6988_I2C_ADDR;
+        defaultAddressApplied = true;
+    }
+
+    busReadRegisterBuffer(busdev, QMP6988_CHIP_ID_REG, &qmp6988_chip_id, 1);  /* read Chip Id */
+
+#if 1
+    if (qmp6988_chip_id != QMP6988_DEFAULT_CHIP_ID) {
+        qmp6988BusDeinit(busdev);
+        if (defaultAddressApplied) {
+            busdev->busdev_u.i2c.address = 0;
+        }
+        return false;
+    }
+#endif
+
+    // IO_SETUP
+    // qmp6988WriteRegister(busdev, QMP6988_IO_SETUP_REG, QMP6988_MODE);
```

Comment:
Why is this commented out?

Reply:
I have change the coding style error.  we merge the code of betaflight only the frist, sorry!

Reply:
No reason to be sorry. We like our code to adhere to good coding standards, but we are happy to give new contributors leeway to get it right. Thank you for your contribution.